src/markupsafe/__init__.py

```diff
@@ -208,6 +214,10 @@ def format(self, *args: t.Any, **kwargs: t.Any) -> "Markup":
         formatter = EscapeFormatter(self.escape)
         return self.__class__(formatter.vformat(self, args, kwargs))
 
+    def format_map(self, map: t.Mapping[str, t.Any]) -> str:  # type: ignore[override]
```

Comment:
Shouldn't this return `Markup`? Although technically it would be even more correct to return `typing_extensions.Self`, since you're using `self.__class__`, but that's a refactor you would have to apply pretty much everywhere.